pb-rs/src/types.rs

```diff
@@ -499,10 +507,35 @@ impl Field {
         Ok(())
     }
 
-    fn write_get_size<W: Write>(&self, w: &mut W) -> Result<()> {
+    fn write_get_size<W: Write>(&self, w: &mut W, desc: &FileDescriptor) -> Result<()> {
         write!(w, "        + ")?;
         let tag_size = sizeof_varint(self.tag());
         match self.frequency {
+            Frequency::Optional
+                if desc.syntax == Syntax::Proto2 || self.typ.message().is_some() =>
+            {
+                // TODO this might be incorrect behavior for proto2
```

Comment:
can you clarify?

Reply:
Hmmm. I wrote this a while ago, so this is what I _think_ I meant:

For proto2, if a field is set, I believe it should be encoded on the wire, even if it is set to the default value. This looks like it sets the size to `0` if the set value matches the default (and I assume it then doesn't encode the default value on the wire). This issue is not new to this PR, I think I included the TODO accidentally.